server/src/main/java/com/thoughtworks/go/server/service/ElasticAgentPluginService.java

```diff
@@ -207,10 +210,16 @@ public String getPluginStatusReport(String pluginId) {
         throw new UnsupportedOperationException("Plugin does not plugin support status report.");
     }
 
-    public String getAgentStatusReport(String pluginId, JobIdentifier jobIdentifier, String elasticAgentId) {
+    public String getAgentStatusReport(String pluginId, JobIdentifier jobIdentifier, String elasticAgentId) throws Exception {
         final ElasticAgentPluginInfo pluginInfo = elasticAgentMetadataStore.getPluginInfo(pluginId);
         if (pluginInfo.getCapabilities().supportsAgentStatusReport()) {
-            return elasticAgentPluginRegistry.getAgentStatusReport(pluginId, jobIdentifier, elasticAgentId);
+            JobPlan jobPlan = jobInstanceSqlMapDao.loadPlan(jobIdentifier.getId());
+            if (jobPlan != null) {
+                ClusterProfile clusterProfile = jobPlan.getClusterProfile();
+                Map<String, String> clusterProfileConfigurations = (clusterProfile == null) ? Collections.emptyMap() : clusterProfile.getConfigurationAsMap(true);
+                return elasticAgentPluginRegistry.getAgentStatusReport(pluginId, jobIdentifier, elasticAgentId, clusterProfileConfigurations);
+            }
+            throw new Exception("Agent has been terminated. Hence, status is no longer available");
```

Comment:
Can we say `Could not fetch agent status report for agent "agent-uuid" as either the job running on the agent has been completed or the agent has been terminated.`